backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java

```diff
@@ -23,73 +23,84 @@ public UserService(UserDb userDb) {
         this.userDb = userDb;
     }
 
-    public List<Game> listAllGames(String userId) {
-        Optional<User> user = userDb.findById(userId);
+    public List<Game> listAllGames(String username) {
+        Optional<User> user = userDb.findByUserName(username);
         if (user.isPresent()) {
             return user.get().getPlayedGames();
         }
         return List.of();
     }
 
-    public List<GameSession> listAllSessions(String userId) {
-        Optional<User> user = userDb.findById(userId);
-        List<GameSession> sessionList = new ArrayList<>();
-        if (user.isPresent()) {
-            List<Game> gameList = user.get().getPlayedGames();
-            if (gameList != null) {
-                for (Game game : gameList) {
-                    if (game != null) {
-                        List<GameSession> existingSessionList = game.getGameSessionList();
-                        for (GameSession session : existingSessionList) {
-                            if (session != null) {
-                                sessionList.add(session);
-                            }
-                        }
-                    }
-                }
-            }
+    public List<GameSession> listAllSessions(String username) {
+        Optional<User> user = userDb.findByUserName(username);
+        if (user.isEmpty()) {
+            return List.of();
         }
-        return sessionList;
+        List<Game> gameList = user.get().getPlayedGames();
+        if (gameList == null) {
```

Comment:
In case of missing gameList, maybe set an empty List within the user model to prevent NPEs?